src/components/ConfirmContent.tsx

```diff
@@ -0,0 +1,161 @@
+import React, {ReactNode} from 'react';
+import {StyleProp, TextStyle, View, ViewStyle} from 'react-native';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import useThemeStyles from '@styles/useThemeStyles';
+import variables from '@styles/variables';
+import Button from './Button';
+import Header from './Header';
+import Icon from './Icon';
+import Text from './Text';
+
+type ConfirmContentProps = {
+    /** Title of the modal */
+    title: string;
+
+    /** A callback to call when the form has been submitted */
+    onConfirm: (...args: unknown[]) => void;
+
+    /** A callback to call when the form has been closed */
+    onCancel: (...args: unknown[]) => void;
+
+    /** Confirm button text */
+    confirmText: string;
+
+    /** Cancel button text */
+    cancelText: string;
+
+    /** Modal content text/element */
+    prompt: string | ReactNode;
+
+    /** Whether we should use the success button color */
+    success: boolean;
+
+    /** Whether we should use the danger button color. Use if the action is destructive */
+    danger: boolean;
+
+    /** Whether we should disable the confirm button when offline */
+    shouldDisableConfirmButtonWhenOffline: boolean;
+
+    /** Whether we should show the cancel button */
+    shouldShowCancelButton: boolean;
+
+    /** Icon to display above the title */
+    iconSource: null | string | ((props: unknown) => ReactNode);
```

Comment:
```suggestion
    iconSource?: (props: SrcProps) => React.ReactNode;
```

Let's change this one a little bit in order to match with `Icon`'s `src` prop. You will also have to import `SvgProps` type from `src/components/Icon/index.tsx`.